storage/rocksdb/ha_rocksdb.cc

```diff
@@ -9983,7 +9983,15 @@ int ha_rocksdb::inplace_populate_sk(const TABLE* new_table_arg,
     {
       DBUG_RETURN(res);
     }
-    tx->commit();
+
+    if (commit_in_the_middle())
+    {
+      release_scan_iterator();
+    }
+    else
+    {
+      tx->commit();
```

Comment:
It seems safe to skip the commit(), but would it be better to always commit() so that existing data is flushed out to rocksdb before we acquire the snapshot and setup the iterator to scan the PK?